server/block/button.go

```diff
@@ -0,0 +1,153 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"time"
+)
+
+// TODO: Activate on projectile hit
+
+// Button is a non-solid block that can provide temporary redstone power.
+type Button struct {
+	empty
+	transparent
+	sourceWaterDisplacer
+
+	// Type is the type of the button.
+	Type ButtonType
+	// Facing is the face of the block that the button is on.
+	Facing cube.Face
+	// Pressed is whether the button is pressed or not.
+	Pressed bool
+}
+
+// FuelInfo ...
+func (b Button) FuelInfo() item.FuelInfo {
+	if b.Type == StoneButton() || b.Type == PolishedBlackstoneButton() {
+		return item.FuelInfo{}
+	}
+	return newFuelInfo(time.Second * 5)
```

Comment:
This should probably be checking if type == Wood() if future types were to be added, although Wood currently takes a parameter...